zcash_primitives/src/sapling/zip32.rs

```diff
@@ -329,7 +350,14 @@ impl ExtendedSpendingKey {
         let depth = reader.read_u8()?;
         let mut tag = [0; 4];
         reader.read_exact(&mut tag)?;
-        let i = reader.read_u32::<LittleEndian>()?;
+        let child_index = reader.read_u32::<LittleEndian>().and_then(|i| {
+            KeyIndex::new(depth, i).ok_or_else(|| {
+                io::Error::new(
+                    io::ErrorKind::Unsupported,
+                    "Non-hardened keys are not supported",
```

Comment:
This error message might not be correct if we make the depth 0, index non-zero case an error.